src/structs/field.ts

```diff
@@ -122,30 +122,41 @@ ${this.isThreadStatic || this.isLiteral ? `` : ` // 0x${this.offset.toString(16)
         }
 
         /** @internal */
-        withHolder(instance: Il2Cpp.Object | Il2Cpp.ValueType): Il2Cpp.Field<T> {
+        withHolder(instance: Il2Cpp.Object | Il2Cpp.ValueType): Il2Cpp.HeldField<T> {
             if (this.isStatic) {
                 raise(`cannot access static field ${this.class.type.name}::${this.name} from an object, use a class instead`);
             }
 
-            const valueHandle = instance.handle.add(this.offset - (instance instanceof Il2Cpp.ValueType ? Il2Cpp.Object.headerSize : 0));
-
-            return new Proxy(this, {
-                get(target: Il2Cpp.Field<T>, property: keyof Il2Cpp.Field): any {
-                    if (property == "value") {
-                        return read(valueHandle, target.type);
-                    }
-                    return Reflect.get(target, property);
-                },
-
-                set(target: Il2Cpp.Field<T>, property: keyof Il2Cpp.Field, value: any): boolean {
-                    if (property == "value") {
-                        write(valueHandle, value, target.type);
-                        return true;
-                    }
-
-                    return Reflect.set(target, property, value);
-                }
-            });
+            return HeldField.from<T>(this, instance);
+        }
+    }
+
+    export class HeldField<T extends Il2Cpp.Field.Type = Il2Cpp.Field.Type> extends Field<T> {
+        /** @internal */
+        constructor(handle: NativePointerValue, public instance: Il2Cpp.Object | Il2Cpp.ValueType) {
+            super(handle);
+        }
+
+        static from<T extends Il2Cpp.Field.Type>(field: Il2Cpp.Field<T>, instance: Il2Cpp.Object | Il2Cpp.ValueType): HeldField<T> {
+            return new HeldField(field.handle, instance);
+        }
```

Comment:
This relates to https://github.com/vfsfitvnm/frida-il2cpp-bridge/pull/598/files#r1948056975.

I think we can just remove this method, as `Il2Cpp.Field<T>::bind` should be used instead

```suggestion
```